dspace/config/spring/api/core-hibernate.xml

```diff
@@ -8,7 +8,17 @@
         <property name="dataSource" ref="dataSource" />
     </bean>
 
-    <bean id="dataSource" class="org.apache.commons.dbcp2.BasicDataSource" lazy-init="true" destroy-method="close">
+    <bean id='dataSource'
+          class='org.springframework.jndi.JndiObjectFactoryBean'>
+        <description>
+            Try to look up the DataSource in JNDI.  If not found, return a
+            DataSource built from connection details in the DSpace configuration.
+        </description>
+        <property name='jndiName' value='java:comp/env/jdbc/dspace'/>
```

Comment:
Just as a note, if we *wanted* to restore `db.jndi`, this line could simply be changed to `value='${db.jndi}'` once we merge #1104 (as it allows any configuration to be referenced from Spring bean definitions).

But, we probably should ask ourselves if we really want to even restore `db.jndi` to the main configurations, or if we only want it to be configurable at the Spring bean layer.